packages/main/src/MenuItem.ts

```diff
@@ -201,6 +230,26 @@ class MenuItem extends ListItem implements IMenuItem {
 	@i18n("@ui5/webcomponents")
 	static i18nBundle: I18nBundle;
 
+	/**
+	 * Defines whether `ui5-menu-item` is in selected state.
+	 *
+	 * **Note:** selected state is only taken into account when `ui5-menu-item` is added to `ui5-menu-item-group`
+	 * with `itemSelectionMode` other than `None`.
+	 * **Note:** A selected `ui5-menu-item` have selection mark displayed ad its end.
+	 * @default false
+	 * @public
+	 * @since 2.5.0
+	 */
+	@property({ type: Boolean })
+	set isSelected(value: boolean) {
+		this.fireDecoratorEvent("item-selection");
```

Comment:
You must only fire events upon user interaction, not when the application sets a property.

Also, custom setters are an anti-pattern, and are only accepted in extreme cases where **synchronous** behavior is needed (f.e. the Popover `opener`/`open` property that must open the popover in the same tick for the iOS soft keyboard to work).

Remove the custom setter, make it a normal property, and fire the event only when the user manually selects/deselects a menu item (and now when the application sets it).